tsdb/shard.go

```diff
@@ -1746,6 +1746,16 @@ func (fs *MeasurementFieldSet) IsEmpty() bool {
 	return len(fs.fields) == 0
 }
 
+func (fs *MeasurementFieldSet) MeasurementFields() map[string]*MeasurementFields {
```

Comment:
I think this is not required for this PR?

Reply:
Correct, this is setup for https://github.com/influxdata/plutonium/issues/3845 which was initially envisioned as an immediate follow-up to this.  Code could be pulled if you wish, but it will go back in a PR very soon.  Your call.

Reply:
I'd prefer to keep each PR self-contained. It's hard to review code without seeing how it is used.